content/docs/tasks/traffic-management/egress-gateway.md

```diff
@@ -0,0 +1,317 @@
+
+Istio 0.8 introduced the concept of [ingress and egress gateways](/docs/reference/config/istio.networking.v1alpha3/#Gateway). Ingress gateways allow one to define entrance points into the service mesh that all incoming traffic flows through. _Egress gateway_ is a symmetrical concept, it defines exit points for the mesh. An egress gateway allows Istio features, for example, monitoring and route rules, to be applied to traffic exiting the mesh.
+
+Another use case is a cluster where the application nodes do not have public IPs, so the in-mesh services that run on them cannot access the Internet. Defining an egress gateway, directing all the egress traffic through it and allocating public IPs to the egress gateway nodes allows the application nodes to access external services in a controlled way.
+
+## Before you begin
+
+* Setup Istio by following the instructions in the
+  [Installation guide](/docs/setup/).
+
+*   Start the [sleep](https://github.com/istio/istio/tree/{{<branch_name>}}/samples/sleep) sample
+    which will be used as a test source for external calls.
+
+    If you have enabled [automatic sidecar injection](/docs/setup/kubernetes/sidecar-injection/#automatic-sidecar-injection), do
+
+    ```command
+    $ kubectl apply -f @samples/sleep/sleep.yaml@
+    ```
+    otherwise, you have to manually inject the sidecar before deploying the `sleep` application:
+
+    ```command
+    $ kubectl apply -f <(istioctl kube-inject -f @samples/sleep/sleep.yaml@)
+    ```
+
+    Note that any pod that you can `exec` and `curl` from would do.
+
+*   Create a shell variable to hold the name of the source pod for sending requests to external services.
+If we used the [sleep](https://github.com/istio/istio/tree/{{<branch_name>}}/samples/sleep) sample, we run:
+    ```command
+    $ export SOURCE_POD=$(kubectl get pod -l app=sleep -o jsonpath={.items..metadata.name})
+    ```
+
+## Define an egress `Gateway` and direct HTTP traffic through it
+
+First let's direct HTTP traffic without TLS origination
+
+1.  Create an egress `Gateway` for _edition.cnn.com_, port 80:
+
+    ```bash
+        cat <<EOF | istioctl create -f -
+        kind: Gateway
+        metadata:
+          name: istio-egressgateway
+        spec:
+          selector:
+            istio: egressgateway
+          servers:
+          - port:
+              number: 80
+              name: http
+              protocol: HTTP
+            hosts:
+            - "edition.cnn.com"
+    EOF
+    ```
+
+1.  Define a `ServiceEntry` for `edition.cnn.com` and a `VirtualService` to direct
+the traffic through the egress gateway:
+
+    ```bash
+        cat <<EOF | istioctl create -f -
```

Comment:
`istioctl create -f` will not work if there are multiple yaml templates, I think we need to use two command to create `ServiceEntry` and `VirtualService`.

Reply:
@gyliu513 It worked for me :)

Reply:
@vadimeisenbergibm Yes, I think you may want to adjust the format in this document.